indexers/keyvalue/MongoDBIndexer/tests/test_mongodbindexer.py

```diff
@@ -1,62 +1,78 @@
-import string
 import random
+from collections import Iterator
 
 import numpy as np
-from google.protobuf.json_format import MessageToJson
+from jina import Document
 
-from jina.drivers.helper import array2pb
-from jina.proto import jina_pb2, uid
 from .. import MongoDBIndexer
 
 
 def random_docs(num_docs, chunks_per_doc=5, embed_dim=10, jitter=1):
     c_id = 3 * num_docs  # avoid collision with docs
     for j in range(num_docs):
-        d = jina_pb2.Document()
-        d.tags['id'] = j
-        d.text = b'hello world doc id %d' % j
-        d.embedding.CopyFrom(array2pb(np.random.random([embed_dim + np.random.randint(0, jitter)])))
-        d.id = uid.new_doc_id(d)
-        for k in range(chunks_per_doc):
-            c = d.chunks.add()
-            c.text = 'i\'m chunk %d from doc %d' % (c_id, j)
-            c.embedding.CopyFrom(array2pb(np.random.random([embed_dim + np.random.randint(0, jitter)])))
-            c.tags['id'] = c_id
-            c.tags['parent_id'] = j
-            c_id += 1
-            c.parent_id = d.id
-            c.id = uid.new_doc_id(c)
+        with Document() as d:
+            d.tags['id'] = j
+            d.text = b'hello world doc id %d' % j
+            d.embedding = np.random.random([embed_dim + np.random.randint(0, jitter)])
+            for k in range(chunks_per_doc):
+                with Document() as c:
+                    c.text = 'i\'m chunk %d from doc %d' % (c_id, j)
+                    c.embedding = np.random.random([embed_dim + np.random.randint(0, jitter)])
+                    c.tags['id'] = c_id
+                    c.tags['parent_id'] = j
+                    c_id += 1
+                    c.parent_id = d.id
+                d.chunks.add(c)
         yield d
 
 
 def test_mongodbindexer():
-    keys = []
-    values = []
-    
     num_docs = 5
-    docs = list(random_docs(num_docs=num_docs, 
+    docs = list(random_docs(num_docs=num_docs,
                             chunks_per_doc=3))
-    keys = [uid.id2hash(doc.id) for doc in docs]
+    # adding type annotations to remove further warnings
+    keys: Iterator[int] = [doc.id for doc in docs]
     values = [doc.SerializeToString() for doc in docs]
-    
+
     query_index = random.randint(0, num_docs - 1)
-    query_id = docs[query_index].id
-    query_key = uid.id2hash(query_id)
+    # adding type annotations to remove further warnings
+    query_key: int = docs[query_index].id
     query_text = docs[query_index].text
 
+    # add
     with MongoDBIndexer() as mongo_indexer:
         mongo_indexer.add(keys=keys, values=values)
-        
+
+    with MongoDBIndexer() as mongo_query:
+        result = mongo_query.query(key=query_key)
+        assert result['_id'] == query_key
+        d = Document()
+        d.ParseFromString(result['values'])
+        assert d.text == query_text
+
+    # update
+    new_docs = list(random_docs(num_docs=num_docs, chunks_per_doc=3))
```

Comment:
Not sure, but that feels like it introduces id collisions. We should switch from explicit `id` definitions to random ids, as soon as it is implemented here: https://github.com/jina-ai/jina/pull/1392

Anyhow, that does not block this PR.